### ISSUE TITLE
Consider publishing the beta releases to NuGet.

### ISSUE THREAD

**Issue by teo-tsirpanis** [fsprojects/FSharp.Formatting#475]
Since that version 3.0.0-beta01 is on NuGet, it gives the impression that there are no other releases after that, despite the fact that they exist on Appeyor, where they are not that easy to find.

**Comment by matthid** [comment]
Yes the workflow is easier on appveyor... but we might need to document it better

**Comment by tpetricek** [comment]
Just to follow-up on this - I was at open F# and two people there asked about our plan to release new non-beta release of F# Formatting on NuGet...

I have not been very closely following the developments, but what do we need before we can do that? One of the two said they're very happily using the beta and have no issues with it, so it sounds like it would be nice to have a new release sometime soon...

**Comment by matthid** [comment]
> but what do we need before we can do that

What exactly? 

* Releasing existing beta releases to nuget is just a matter of "doing it" the good thing about appveyor is that I can "just" commit and don't need to run the release process locally (and is therefore faster).
* Releasing a "stable" version is another subject, there are a lot of ideas floating around but nothing has been finished yet. Minimal requirement (imho) for a stable v3 release is to support netcore/netstandard-dlls as well as running on netcore/netstandard. Ideally we use the breaking change to no longer bundle "razor" and instead use something else by default.

**Comment by wallymathieu** [comment]
I could help with publishing beta to nuget if needed.

**Comment by tpetricek** [comment]
Thanks for the quick reply @matthid and for the offer @wallymathieu!

I got the impression (from what I heard at open F#) that netcore/netstandard was already working quite well?

As for removing Razor dependency, I completely support that - though perhaps we could get away with not replacing it with anything initially and letting people add whatever they want as a separate component? (Replacing it completely sounds like quite a lot of work...).

**Comment by thinkbeforecoding** [comment]
Releasing only fsharp.literate as a nuget for netstandard2 would be awesome.
Then using fable react templating…

jeremie chassaing / thinkbeforecoding

De : Tomas Petricek
Envoyé le :mardi 2 octobre 2018 15:38
À : fsprojects/FSharp.Formatting
Cc : Subscribed
Objet :Re: [fsprojects/FSharp.Formatting] Consider publishing the betareleases to NuGet. (#475)

Thanks for the quick reply @matthid and for the offer @wallymathieu!
I got the impression (from what I heard at open F#) that netcore/netstandard was already working quite well?
As for removing Razor dependency, I completely support that - though perhaps we could get away with not replacing it with anything initially and letting people add whatever they want as a separate component? (Replacing it completely sounds like quite a lot of work...).

**Comment by matthid** [comment]
> netcore/netstandard was already working quite well?

Afaik this is because multi targetting is so easy and people just use 'TargetFrameworks>net461; netstandard2.0<'. Also we do not run on netstandard yet (which means you cannot use it within fake5 or without mono)

Also the razor part is not a blocker I'd really recommend to get rid of it. Otherwise someone would need to do the work of porting RazorEngine to netstandard as well...

> Releasing only fsharp.literate as a nuget for netstandard2 would be awesome

@thinkbeforecoding While I'm open for changes like this you would have to send a PR to make that happen as some infrastructure has to be changed to publish that independently. And we have no plans to do it ;)

**Comment by kurtschelfthout** [comment]
Short term I strongly support releasing betas to nuget as they are.

**Comment by wallymathieu** [comment]
@matthid would it be OK to put the beta packages on nuget?

**Comment by dsyme** [comment]
Since ProjectScaffold is already picking up the beta Nuget packages from an unofficial source I really think we should just remove the "beta" tag and always publish to nuget.

This cat is already out of the bag - people create project scaffold projects all the time.....

**Comment by matthid** [comment]
From my side, nothing has changed since [this comment](https://github.com/fsprojects/FSharp.Formatting/issues/475#issuecomment-426267365).
So it's just a matter of someone either doing the work manually or automating the process.

> @matthid would it be OK to put the beta packages on nuget?

Sure
